specifyweb/frontend/js_src/lib/components/WbImportAttachments/index.tsx

```diff
@@ -0,0 +1,220 @@
+/**
+ * Workbench Attachment import page
+ *
+ * @module
+ */
+
+import React from 'react';
+import { useNavigate } from 'react-router-dom';
+
+import { ajax } from '../../utils/ajax';
+import { attachmentsText } from '../../localization/attachments';
+import { commonText } from '../../localization/common';
+import type { RA } from '../../utils/types';
+import { Container, H2 } from '../Atoms';
+import { Progress } from '../Atoms';
+import { Button } from '../Atoms/Button';
+import { uploadFile } from '../Attachments/attachments';
+import {
+  deserializeResource,
+  serializeResource,
+} from '../DataModel/serializers';
+import { tables } from '../DataModel/tables';
+import type { SpDataSetAttachment } from '../DataModel/types';
+import { raise } from '../Errors/Crash';
+import { useMenuItem } from '../Header/MenuContext';
+import { userInformation } from '../InitialContext/userInformation';
+import { loadingBar } from '../Molecules';
+import { Dialog } from '../Molecules/Dialog';
+import { FilePicker } from '../Molecules/FilePicker';
+import { Preview } from '../Molecules/FilePicker';
+import { uniquifyDataSetName } from '../WbImport/helpers';
+import { ChooseName } from '../WbImport/index';
+import type { SerializedRecord } from '../DataModel/helperTypes';
+import { f } from '../../utils/functools';
+import { useBooleanState } from '../../hooks/useBooleanState';
+
+export function WbImportAttachmentsView(): JSX.Element {
+  useMenuItem('workBench');
+  const [files, setFiles] = React.useState<readonly File[] | undefined>();
+
+  return (
+    <Container.Full>
+      <H2>{commonText.multipleFilePickerMessage()}</H2>
+      <div className="w-96">
+        <FilePicker
+          acceptedFormats={undefined}
+          showFileNames
+          onFilesSelected={(selectedFiles) => {
+            setFiles(Array.from(selectedFiles));
+          }}
+        />
+      </div>
+      {files !== undefined && files.length > 0 && <FilesPicked files={files} />}
+    </Container.Full>
+  );
+}
+
+function FilesPicked({
+  files,
+}: {
+  readonly files: readonly File[];
+}): JSX.Element {
+  const navigate = useNavigate();
+  const [fileUploadProgress, setFileUploadProgress] = React.useState<
+    number | true | undefined
+  >(undefined);
```

Comment:
Also, I wouldn't recommend having states whose type can vary a lot. It can become difficult to keep track of what to change the state to.
What do you think of restricting `fileUploadProgress` to only be a number and initialize it to 0?